src/logging.c

```diff
@@ -0,0 +1,151 @@
+// Copyright 2017 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#if __cplusplus
+extern "C"
+{
+#endif
+
+#include "rcutils/allocator.h"
+#include "rcutils/logging.h"
+
+bool g_rcl_logging_initialized = false;
+
+rcl_logging_output_handler_t g_rcl_logging_output_handler = NULL;
+
+int g_rcl_logging_severity_threshold = 0;
```

Comment:
It seems to me that all of these are either implementation details (initialized), or already have getters and setters (output_handler, threshold), so we could get rid of the extern on all of them and make them static here.